pkg/client/supplier/client.go

```diff
@@ -82,6 +84,14 @@ func (sClient *supplierClient) SubmitProof(
 		return err
 	}
 
+	// TODO_IMPROVE: log details related to what & how much is being proven
+	logger.Info().
+		Str("supplier_addr", sClient.signingKeyAddr.String()).
+		Str("app_addr", sessionHeader.ApplicationAddress).
+		Str("session_id", sessionHeader.SessionId).
+		Str("service", sessionHeader.Service.Id).
+		Msg("created a new proof")
```

Comment:
With so many fields, it may improve readability and maintainability to use `#Fields()` instead:

```suggestion
	logger.Info().
		Fields(map[string]any{
				"supplier_addr": sClient.signingKeyAddr.String(),
				"app_addr": sessionHeader.ApplicationAddress,
				"session_id": sessionHeader.SessionId,
				"service": sessionHeader.Service.Id,
		}).
		Msg("created a new proof")